src/libopensc/card-piv.c

```diff
@@ -493,6 +497,7 @@ static const struct piv_object piv_objects[] = {
 			"2.16.840.1.101.3.7.2.9999.119", 2, "\x94\x06", "\x94\x06", PIV_OBJECT_TYPE_PUBKEY},
 	{ PIV_OBJ_9506, "Pub 95 key ",
 			"2.16.840.1.101.3.7.2.9999.120", 2, "\x95\x06", "\x95\x06", PIV_OBJECT_TYPE_PUBKEY},
+	{ PIV_OBJ_PRIV, "Priv key ", "", 0, "", "\x50\x15", 0},
```

Comment:
This is not a specific key object but any. When using a key, e.g. for computing a signature, the private key is selected by `sc_pkcs15_*`. This fails without this modification that enables to select the application DF by its path 5015.

Reply:
This sounds like a hackish approach. If I'm not mistaken, then PIV doesn't have any DFs and the PIV AID is selected in piv_init so that the application will be active when reading the applet's data.

Reply:
I changed this to a Swissbit specific path from the card's profile. In order to perform admin operations like key generation I need to have a profile for emulation of a PKCS#15 structure for my non-PKCS#15 PIV card. So I do have DFs. This extension is required to select the private key EF in the key-domain template in the app DF, for instance, during signing with the private key.

Reply:
Removed PIV object for private keys. See comments below.